Argo/Globals/Parser.swift

```diff
@@ -0,0 +1,29 @@
+public enum Parser<T> {
+  case Success(Box<T>)
+  case TypeMismatch(String)
+  case MissingKey(String)
+
+  public var value: T? {
+    switch self {
+    case let .Success(box): return box.value
+    default: return .None
+    }
+  }
+}
+
+public func typeMismatch<T>(type: String, object: JSON) -> Parser<T> {
```

Comment:
But we'd need this to be `public` in order to let users write more complex parsers. Like, for example, the ability to write parsers for `NSURL` (like we did in the tests).